cmd/thanos/main.go

```diff
@@ -143,8 +145,16 @@ func main() {
 	{
 		ctx := context.Background()
 
-		var closeFn func() error
-		tracer, closeFn = tracing.NewOptionalGCloudTracer(ctx, logger, *gcloudTraceProject, *gcloudTraceSampleFactor, *debugName)
+		var closer io.Closer
+		var confContentYaml []byte
+		confContentYaml, err = tracingConfig.Content()
+		//tracer, closer = tracingFactory.Create(ctx, logger, *debugName)
+		tracer, closer, err = provider.NewTracer(ctx, logger, confContentYaml)
```

Comment:
We need to manually make sure `closer` is closed on error cases of this function